File: src/types/equipment/item/MusicalInstrument.ts
import { Entity, IncludeIdentifier, Object, Optional, Required } from "tsondb/schema/def"
import { src } from "../../source/_PublicationRef.js"
import { CombatUse, Complexity, Cost, DefaultItemTranslations, Weight } from "./_Item.js"

export const MusicalInstrument = Entity(import.meta.url, {
  name: "MusicalInstrument",
  namePlural: "MusicalInstruments",
  type: () =>
    Object({
      cost: Required({
        comment: "The cost in silverthalers.",
        type: IncludeIdentifier(Cost),
      }),
      weight: Required({
        comment: "The weight in kg.",
        type: IncludeIdentifier(Weight),
      }),
      complexity: Optional({
        comment: "The complexity of crafting the item.",
        type: IncludeIdentifier(Complexity),
      }),
      combat_use: Optional({
        comment:
          "The item can also be used either as an improvised weapon or as an armor, although this is not the primary use case of the item.",
        type: IncludeIdentifier(CombatUse),
      }),
      src,
      translations: DefaultItemTranslations("MusicalInstrument"),
    }),
  displayName: {},
})

File: Sources/OptolithDatabaseSchema/Types/equipment/item/MusicalInstrument.swift
import FileDB

@Model
public struct MusicalInstrument {

  /// The cost in silverthalers.
  @Relationship(Cost)
  let cost: Cost.ID

  /// The weight in kg.
  @Relationship(Weight)
  let weight: Weight.ID
  /// The complexity of crafting the item.
  @Relationship(Complexity)
  let complexity: Complexity.ID?
  /// The item can also be used either as an improvised weapon or as an armor, although this is not the primary use case of the item.
  @Relationship(CombatUse)
  let combat_use: CombatUse.ID?

    /// The publications where you can find the entry.
    let src: PublicationRefs
      translations: DefaultItemTranslations("MusicalInstrument"),
    }),
}
